reaktive/src/commonMain/kotlin/com/badoo/reaktive/observable/WithLatestFrom.kt

```diff
@@ -0,0 +1,82 @@
+package com.badoo.reaktive.observable
+
+import com.badoo.reaktive.base.ErrorCallback
+import com.badoo.reaktive.completable.CompletableCallbacks
+import com.badoo.reaktive.disposable.CompositeDisposable
+import com.badoo.reaktive.disposable.Disposable
+import com.badoo.reaktive.utils.Uninitialized
+import com.badoo.reaktive.utils.lock.newLock
+import com.badoo.reaktive.utils.lock.synchronized
+import kotlin.jvm.JvmName
+
+fun <T, O, R> Observable<T>.withLatestFrom(
```

Comment:
I believe it would be good to implement firstly `fun <T, U, R> Observable<T>.withLatestFrom(others: Collection<Observable<U>>, mapper: (T, List<U>) -> R): Observable<R>` and then just add a number of specific operators (with one, two, etc. other sources).

Reply:
Although they share the same operator name, that is technically a `WithLatestFromMany` and requires a different implementation (see the [RxJava source](https://github.com/ReactiveX/RxJava/blob/2.x/src/main/java/io/reactivex/internal/operators/observable/ObservableWithLatestFromMany.java)). If I were to create it, I would like to do that in another PR.

Reply:
I'm not sure why is it like this in RxJava2 for `withLatestFrom` with one other source, but for `withLatestFrom` with two or more "others" they are using generic version of withLatestFrom (with array of sources).

We can do better, this should work :
```
fun <T, U, R> Observable<T>.withLatestFrom(others: Collection<Observable<U>>, mapper: (T, List<U>) -> R): Observable<R> =
    ...

fun <T, U, R> Observable<T>.withLatestFrom(other: Observable<U>, combine: (value: T, other: U) -> R): Observable<R> =
    withLatestFrom(listOf(other)) { item, others ->
        combine(item, others[0])
    }
```

Reply:
Indefinitely it will work, but when comparing the two ("other" vs. "others") it appears that the "others" variant requires more (expensive) logic. Also it feels inefficient to wrap "other" in a list and later unwrap its result to call the `combine` callback. 

Although opinionated: The "others" variant is not so valuable, because in practice it rarely happens that there are "others" and when it is, their type differ from each other.

Therefore I propose we develop the "others" variant in a different PR.